### DISCUSSION TITLE
*-PowerAppManagementApp equivalents in PAC CLI

### DISCUSSION THREAD

**Discussion by YannickRe** [microsoft/powerplatform-build-tools#479] [Ideas]
After the Office Hours today (and my questions there), I learned that Service Principals can be used to do tenant level API calls but it needs to be registered as a Management Application.
The article: https://learn.microsoft.com/en-us/power-platform/admin/powershell-create-service-principal?WT.mc_id=M365-MVP-5003400
The problems with the article:

- The title says "Create Service Principal" but it never creates a service principal 😀
- It's buried deep in the docs
- It hurries over the "Management Application" concept

Ideally, some of the above things get an upgrade so it can be more widespread knowledge.
And additionally, I'd like the following commands to be translated into PAC CLI so I don't need PowerShell anymore 😀:

- https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/get-powerappmanagementapp?view=pa-ps-latest&WT.mc_id=M365-MVP-5003400
- https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/new-powerappmanagementapp?view=pa-ps-latest&WT.mc_id=M365-MVP-5003400
- https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/get-powerappmanagementapps?view=pa-ps-latest&WT.mc_id=M365-MVP-5003400
- https://learn.microsoft.com/en-us/powershell/module/microsoft.powerapps.administration.powershell/remove-powerappmanagementapp?view=pa-ps-latest&WT.mc_id=M365-MVP-5003400

Those would really help moving people to SPN auth for most actions.

**Comment by snizar007** [comment]
We will look into these.

**Comment by Laskewitz** [comment]
📰 Looks like we have some news here!

Check-out the following links:
[https://aka.ms/pac/admin#pac-admin-application-register](https://aka.ms/pac/admin#pac-admin-application-register)
[https://aka.ms/pac/admin#pac-admin-application-unregister](https://aka.ms/pac/admin#pac-admin-application-unregister)
